src/main/java/org/medicmobile/webapp/mobile/MedicAndroidJavascript.java

```diff
@@ -328,6 +337,8 @@ public void onDateSet(DatePicker view, int year, int month, int day) {
 		dialog.show();
 	}
 
+	@SuppressFBWarnings("DM_DEFAULT_ENCODING")
+	@SuppressWarnings("PMD.CloseResource")
```

Comment:
Instead of suppressing this warning, it is probably better to wrap everything in a try-with-resources block so that both the readers will always be closed (even if an exception gets thrown):

```.java
		try(
			FileReader fr = new FileReader("/proc/cpuinfo");
			BufferedReader bufferedReader = new BufferedReader(fr);
		){
			String line;
			HashMap output = new HashMap();
			while ((line = bufferedReader.readLine()) != null) {
				String[] data = line.split(":");
				if (data.length > 1) {
					String key = data[0].trim();
					if (key.equals("model name")) {
						output.put(key, data[1].trim());
						break;
					}
				}
			}

			int cores = Runtime.getRuntime().availableProcessors();
			output.put("cores", cores);

			String arch = System.getProperty("os.arch");
			output.put("arch", arch);

			return output;
		}
```

Reply:
Nice done!

Reply:
I also got another error in spotbugs: `Found reliance on default encoding`
So end up making the code like from this:
```
FileReader fr = new FileReader("/proc/cpuinfo");
```
to:
```
Reader fileReader = new InputStreamReader(new FileInputStream("/proc/cpuinfo"), StandardCharsets.UTF_8);
			BufferedReader bufferedReader = new BufferedReader(fileReader);
```

Reply:
I think you still need to close the resources in a `finally` block. In other words, we shouldn't just ignore this error.

Reply:
Oh I forgot to remove the suppress, this is fixed. When opening the resources in the `try()` they'll get closed automatically without needing the `finally`.